databroker/core.py

```diff
@@ -819,7 +819,7 @@ def _repr_pretty_(self, p, cycle):
         try:
             self._load()
             start = self.metadata['start']
-            stop = self.metadata['stop']
+            stop = self.metadata['stop'] or {}
```

Comment:
If there's really no stop document, won't this raise a key error instead of returning {}?

Reply:
The `metadata` dict is initialized with `'stop'` set to `None` in `_load()` (which is called at `__init__` time) if the stop doc does not exist yet. It's arguable that leaving it empty would be better, but that would be a backward-incompatible change.


https://github.com/bluesky/databroker/blob/6b7e31b6cc399686df926fec51e43193d488791f/databroker/core.py#L987-L1001

Reply:
I see, cool.